api/src/main/java/org/openmrs/Cohort.java

```diff
@@ -208,15 +210,29 @@ public boolean removeMembership(CohortMembership cohortMembership) {
 	public Collection<CohortMembership> getActiveMemberships() {
 		return getActiveMemberships(new Date());
 	}
-	
+
 	/**
+	 * Searches for the {@code patients} membership and returns it.
+	 * <p>
+	 * If the patient has no membership, an empty Optional is returned.
+	 *
+	 * @param patient Holder of the searched membership.
+	 * @return Optional membership of the given patient.
+	 * @throws NoSuchElementException Thrown if the patient has no active membership.
 	 * @since 2.1.0
 	 */
 	public CohortMembership getActiveMembership(Patient patient) {
-		return getMemberships().stream().filter(m -> m.isActive() && m.getPatientId().equals(patient.getPatientId()))
-		        .collect(Collectors.toList()).get(0);
+		Predicate<CohortMembership> patientMatcher =
+			membership -> membership.getPatientId().equals(patient.getPatientId());
+
+		return getMemberships().stream()
+			.filter(CohortMembership::isActive)
+			.filter(patientMatcher)
+			.findAny()
+			.orElseThrow(NoSuchElementException::new); // Same as calling get()
```

Comment:
Are you sure throwing the above Exception won't cause problems?

Reply:
Calling the `get()` method will also result
in an `NoSuchElementException` to be thrown.
The best thing would be to return the Optional
but I didn't want to make changes to the public API.